plugins/outputs/cloudwatchlogs/pusher.go

```diff
@@ -297,38 +308,103 @@ func (p *pusher) send() {
 			p.Log.Errorf("Aws error received when sending logs to %v/%v: %v", p.Group, p.Stream, awsErr)
 		}
 
-		wait := retryWait(retryCount)
+		// retry wait strategy depends on the type of error returned
+		var wait time.Duration
+		if chooseRetryWaitStrategy(err) == retryLong {
+			wait = retryWaitLong(retryCountLong)
+			retryCountLong++
+		} else {
+			wait = retryWaitShort(retryCountShort)
+			retryCountShort++
+		}
+
 		if time.Since(startTime)+wait > p.RetryDuration {
-			p.Log.Errorf("All %v retries to %v/%v failed for PutLogEvents, request dropped.", retryCount, p.Group, p.Stream)
+			p.Log.Errorf("All %v retries to %v/%v failed for PutLogEvents, request dropped.", retryCountShort+retryCountLong-1, p.Group, p.Stream)
 			p.reset()
 			return
 		}
 
-		p.Log.Warnf("Retried %v time, going to sleep %v before retrying.", retryCount, wait)
+		p.Log.Warnf("Retried %v time, going to sleep %v before retrying.", retryCountShort+retryCountLong-1, wait)
 
 		select {
 		case <-p.stop:
-			p.Log.Errorf("Stop requested after %v retries to %v/%v failed for PutLogEvents, request dropped.", retryCount, p.Group, p.Stream)
+			p.Log.Errorf("Stop requested after %v retries to %v/%v failed for PutLogEvents, request dropped.", retryCountShort+retryCountLong-1, p.Group, p.Stream)
 			p.reset()
 			return
 		case <-time.After(wait):
 		}
 
-		retryCount++
 	}
 
 }
 
-func retryWait(n int) time.Duration {
+// retryWaitShort returns a duration to wait before retrying a request.
+// This implements an exponential backoff strategy with jitter.
+// The base wait time is 200 milliseconds, and the maximum wait time is 1 minute (jittered).
+// The wait time is capped at 1 minute if the retry count exceeds 5.
+func retryWaitShort(retryCount int) time.Duration {
 	const base = 200 * time.Millisecond
 	// Max wait time is 1 minute (jittered)
 	d := 1 * time.Minute
-	if n < 5 {
-		d = base * time.Duration(1<<int64(n))
+	if retryCount < 5 {
+		d = base * time.Duration(1<<int64(retryCount))
+	}
+	return time.Duration(seededRand.Int63n(int64(d/2)) + int64(d/2))
+}
+
+// retryWaitLong returns a duration to wait before retrying a request.
+// This implements an exponential backoff strategy with jitter.
+// The base wait time is 2 seconds, and the maximum wait time is 1 minute (jittered).
+// The wait time is capped at 1 minute if the long-retry count exceeds 2.
+func retryWaitLong(retryCount int) time.Duration {
```

Comment:
nit: Both functions are identical with just different base wait times and retry counts. Couldn't we just pass those in?

Reply:
Yeah, we could parametrize it. I felt the extra layer of abstraction wasn't worth it for these two functions (and I don't anticipate we'd be growing the number of strategies) so I left them separate.